### PR TITLE
fix(ARCH-662/scripts-config-eslint): do not JSON.parse twice

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

same as #4546 

**What is the chosen solution to this problem?**

* feat(config-jest):  add __tests__ folder support like with default jest config
* fix(config-eslint): do not parse the config twice it throw an error
* feat(scripts-core): log error of the scripts if there is a JS error
* chore(scripts-core): use `@talend/scripts-utils` package and move most of utils here
* add tests on all theses packages

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: tools/scripts-core/utils/preset.js [hunk 1]
```diff
@@ -18,48 +18,4 @@ export function getPresetApi(env = process.env) {
...
-		isLib: !isApp,
-		isApp,
-		isReact,
-		isAngular,
-		isPublic,
-	};
-}
-
-export function getConfigurationPackage() {}
+export function getPresetEnv() {}
```
Comment (smouillour) [inline tools/scripts-core/utils/preset.js:21]: It looks like not used. maybe we can remove it 🤔

File: tools/scripts-core/package.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
     "url": "https://github.com/Talend/ui.git"
   },
   "scripts": {
-    "lint": "talend-scripts lint ./",
+    "lint": "talend-scripts lint ./scripts/",
```
Comment (smouillour) [inline tools/scripts-core/package.json:21]: no lint in utils folder?

Comment (jmfrancois) [reply tools/scripts-core/package.json:21]: I have moved all file under src so I can use without arguments

**Review by smouillour — APPROVED** [review]

**Review by lmaillet — APPROVED** [review]
I hope it will fix my issue with `@talend\eslint-config` 😉

**Pull request merged**